scripts/sanity_check_test.py

```diff
@@ -0,0 +1,139 @@
+import unittest
+import sys
+
+import sanity_check
+import git
+
+class ValidateTests(unittest.TestCase):
+    """ Tests for commit validation """
+
+    def assertFailure(self, commits, expected):
+        from StringIO import StringIO
+
+        real_commits = convert_commits(commits)
+        out = StringIO()
+
+        self.assertRaises(SystemExit, sanity_check.validate_commits, real_commits, out)
+        assert out.getvalue().strip() == expected
+
+    def assertSuccess(self, commits):
+        real_commits = convert_commits(commits)
+
+        try:
+            sanity_check.validate_commits(real_commits)
+        except:
+            print "Unexpected error:", sys.exc_info()[0]
+            raise
+
+
+    # Test cases that excercise invalid scenarios.
+
+    def test_squash_not_matching_carry(self):
+        """
+        [SQUASH] commit has a type that is not matching any [CARRY].
+        """
+
+        commits = [
```

Comment:
Enforcing intent in our carries is a good way to maintain as few as possible so I think there is value in cross-commit validation (why would I submit a SQUASH w/o a matching CARRY?)